netket/graph/lattice.py

```diff
@@ -104,323 +107,349 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
-            ]
-        )
-        if (
-            atoms_coord_fractional.min() < -cutoff_tol
-            or atoms_coord_fractional.max() > 1 + cutoff_tol
-        ):
-            # Maybe there is another way to state this. I want to avoid that there exists the possibility that two atoms from different cells are at the same position:
-            raise ValueError(
-                "atoms must reside inside their corresponding unit cell, which includes only the 0-faces in fractional coordinates."
-            )
-        uniques = _np.unique(atoms_coord, axis=0)
-        if len(atoms_coord) != uniques.shape[0]:
-            atoms_coord = _np.asarray(uniques)
-            warnings.warn(
-                f"Some atom positions are not unique. Duplicates were dropped, and now atom positions are {atoms_coord}",
-                UserWarning,
-            )
+        self._primitives = self._clean_primitives(primitives)
+        self._ndim = self._primitives.shape[1]
 
-        self._atoms_coord = atoms_coord
+        self._basis, basis_fractional = self._clean_basis(basis, self._primitives)
+        self._pbc = self._clean_pbc(pbc, self._ndim)
 
-        if isinstance(pbc, bool):
-            self._pbc = [pbc] * self._basis_vectors.shape[1]
-        elif (
-            not isinstance(pbc, list)
-            or len(pbc) != self._basis_vectors.shape[1]
-            or sum([1 for pbci in pbc if isinstance(pbci, bool)])
-            != self._basis_vectors.shape[1]
-        ):
-            raise ValueError(
-                "pbc must be either a boolean or a list of booleans with the same dimension as the vector space you chose."
-            )
-        else:
-            self._pbc = pbc
+        self.extent = _np.asarray(extent, dtype=int)
 
-        extent = _np.asarray(extent)
-        self.extent = extent
-
-        atoms, cellANDlabel_to_site = create_points(
-            self._basis_vectors, extent, atoms_coord_fractional, pbc
+        sites, self._cell_coord_to_site = create_sites(
+            self._primitives, self.extent, basis_fractional, pbc
         )
         edges = get_true_edges(
-            self._basis_vectors, atoms, cellANDlabel_to_site, extent, distance_atol
+            self._primitives,
+            sites,
+            self._cell_coord_to_site,
+            self.extent,
+            distance_atol,
         )
         graph = _nx.MultiGraph(edges)
 
-        # Rename atoms
-        old_nodes = sorted(set([node for edge in edges for node in edge]))
-        self._atoms = [atoms[old_node] for old_node in old_nodes]
-        self._coord_to_site = {
-            tuple(atom["r_coord"]): new_site
-            for new_site, atom in enumerate(self._atoms)
-        }
+        # Rename sites
+        old_nodes = sorted(set(node for edge in edges for node in edge))
+        self._sites = []
+        for i, site in enumerate(sites[old_node] for old_node in old_nodes):
+            site.id = i
+            self._sites.append(site)
```

Comment:
```suggestion
        self._sites = sites 
```
Or even better, let `create_sites` return `self._sites` directly

Reply:
I suspect some of the next thirteen lines can also be dropped, I just can't select them

Reply:
See https://github.com/netket/netket/pull/715#discussion_r633422781